toolsWebsite/tools/views.py

```diff
@@ -71,13 +71,64 @@ def handle_uploaded_file(f):
     with open(f"tools/static/images/{f}", 'wb+') as destination:
         for chunk in f.chunks():
             destination.write(chunk)
+
+def checkout(request):
+    # We can do all sorts of stuff with an html template with the cart info
+    # that will be stored in our cart models.  I.e. tell the customer when it will be
+    # due/ do a confirmation page etc.
+
+    for cart in ShoppingCart.objects.all():
+        due = DueDates()
+        due.toolCategory = cart.toolCategory
+        due.quantity = cart.quantity
+        due.date_bought = datetime.datetime(2019, randint(1, 12), randint(1, 28))
+        due.date_due = datetime.datetime(2020, randint(1, 12), randint(1, 28))
+        # due.date_bought = datetime.datetime.now()
+        # due.date_due = datetime.datetime.now()
+        due.buyer = "Connor"
+        due.save()
+        cart.delete()
+
+
+    return render(request, 'tools/checkout.html')
+
+
+def addToCart(request, category_id):
+    tool = get_object_or_404(ToolCategory, pk=category_id)
+    tool.available -= 1
+    tool.unavailable += 1
+    tool.save()
+
+    inCart = False
+    for cart in ShoppingCart.objects.all():
+        if cart.tool == tool.type:
+            cart.toolCategory = tool
+            cart.tool = tool.type
```

Comment:
if the tool is already in the shopping cart would we need to save the toolcategory and tool again? couldn't we just increment the quantity and then save?

Reply:
Deleted those lines.